modules/core/src/main/java/org/apache/ignite/internal/util/IgniteUtils.java

```diff
@@ -5380,6 +5380,37 @@ public static void writeMap(ObjectOutput out, Map<?, ?> map) throws IOException
         return map;
     }
 
+
+    /**
+     * Calculate a hashCode for an array.
+     *
+     * @param obj Object.
+     */
+    public static int hashCode(Object obj) {
+        if (obj.getClass().isArray()) {
+            if (obj instanceof byte[])
+                return Arrays.hashCode((byte[])obj);
+            if (obj instanceof short[])
+                return Arrays.hashCode((short[])obj);
+            if (obj instanceof int[])
+                return Arrays.hashCode((int[])obj);
+            if (obj instanceof long[])
+                return Arrays.hashCode((long[])obj);
+            if (obj instanceof float[])
+                return Arrays.hashCode((float[])obj);
+            if (obj instanceof double[])
+                return Arrays.hashCode((double[])obj);
+            if (obj instanceof char[])
+                return Arrays.hashCode((char[])obj);
+            if (obj instanceof boolean[])
+                return Arrays.hashCode((boolean[])obj);
+
+            return Arrays.hashCode((Object[])obj);
```

Comment:
Seems that this won't work properly if the key is array of arrays.

Reply:
Yes. I suppose it's expected:)

Reply:
I fixed it. Added deep hash code.